File: Ehr_Backend/Ehr_Backend/src/main/java/Entity/EHR.java
package Entity;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Document(collection = "ehr")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class EHR {

    @Id
    private String id;
    private String patientId; // Use String for reference (ObjectId)
    private String doctorId;
    private Date date;
    private String diagnosis;
    private String treatment;
    private String notes;

    // Getters and Setters
}

File: EHR_node_Backend/models/EHR.js
const mongoose = require('mongoose');

const ehrSchema = new mongoose.Schema({
    patientId: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Patient',
        required: true
    },
    doctorId: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Doctor',
        required: true
    },
    date: {
        type: Date,
        default: Date.now
    },
    diagnosis: {
        type: String,
        required: true
    },
    treatment: {
        type: String,
        required: true
    },
    notes: String
}, {
    timestamps: true
});

module.exports = mongoose.model('EHR', ehrSchema); 